### DISCUSSION TITLE
"Push to get suggestion"

### DISCUSSION THREAD

**Discussion by akaKarthick** [czbar/ChessForge#649] [Ideas]
Hi, 

I have a few suggestions for the **"push to get suggestion"** feature.

**First,** I think it would be helpful to have the option to see only one move in the line, rather than the entire line. This would allow users to get the computer's opinion on a specific move, without being influenced by the other moves in the line. (**Optional**)

**Second**, I think it would be beneficial to have a button that users can press to get the computer's evaluation of a position, without any move suggestions. This would allow users to focus on the position and make their own decisions, while still getting the benefit of the computer's analysis.

The button could be labeled **"Get Evaluation**" or something similar, and it could be placed in a convenient location, such as the toolbar or the status bar. When the button is pressed, the evaluation bar would appear, but there would be no move suggestions. This would allow users to get the computer's opinion on the position without being influenced by the move suggestions.

I believe that these features would make the **"push to get suggestion"** feature more versatile and useful for users. They would allow users to get the benefit of the computer's analysis, while still maintaining their own decision-making process.

I think the "Get Evaluation" feature would be most suitable in the **Exercise** mode, **Study** window. The first one i mentioned would be exclusive for **Exercise**. 

What is your thought about this?

**Comment by czbar** [comment]
Let me paraphrase and clarify these suggestions. Correct me if I got some of your ideas wrong:

**Item 1**: You would like to see this feature in the **Exercise view** while in the **Solving mode**. Is that right?  If so, it would only apply in "_Enter your analysis and submit_" (since it would make no sense in the "_Guess the next move_" mode).

And then, when you say "_only one move in the line, rather than the entire line_", do you mean to show it in the list of engine moves under the chessboard?

And would that be just the best move (so, only move would be shown) or a list of moves, like there is a list of lines now?

**Item 2**: Here, when you say "_Exercise mode, Study window_" do you mean again the **Solving** mode with "_Enter your analysis and submit_"?

**Reply by akaKarthick** [reply]
You almost got it. Now i changed little bit after some thought.

There are two feature in my mind.

**1. "Get evaluation"
2. " Suggest Move"**

**Item 1**: When i Press the `Ctrl`+`?` , the eval bar should pop up and show the engine's opinion on the position. The move suggestions won't be shown.
This feature is most useful in the Exercise (**"Enter your analysis and submit"**) and **Study** window.

**Item 2**: Similar to the "Get evaluation" feature, but this time only the best move is shown, **not the entire line** and **eval bar**.
This feature is most useful in the Exercise ("**Enter your analysis and submit**") and **Study** window. 
It could be like Arrows.

![Screenshot (363)](https://github.com/czbar/ChessForge/assets/125844120/3f2f8a8c-2fbc-47dc-aa59-4a9b498fbec7)

Here are some possible names for the features:

**"Get evaluation"
"Engine's opinion"
"Best move"
"Suggested move"**

And here are some possible button assignments:

**Ctrl+?
Alt+?
F1
Spacebar**
I think it would be a good idea to test out different names and button assignments to see what works best. You could get feedback from other chess players to see what they think.

What do you think about this? is it worth to implement? It just some thought in my mind, You can make it perfect.

**Reply by czbar** [reply]
I like the idea of the arrow showing the best move, like the web sites do. Ok, I'll return to these ideas later on.

**Comment by czbar** [comment]
I have opened an issue based on this discussion.  I always keep an eye on issues but tend to forget about discussions.
